src/context/mod.rs

```diff
@@ -359,7 +421,14 @@ impl<'a> QueryMatchContext<'a> {
         self.file_run_context.language
     }
 
-    // pub fn retrieve<TFromFileRunContext: FromFileRunContext>(&self) -> TFromFileRunContext {}
+    pub fn retrieve<TFromFileRunContext: FromFileRunContext<'a> + TidAble<'a>>(
+        &self,
+    ) -> &TFromFileRunContext {
```

Comment:
So per above comments I guess I haven't actually tried eg storing a provided-type instance in per-file-run rule state yet

This return type of `&` (vs eg `&'a`) makes me think that it won't let me do that because it's going to treat its lifetime as that of the `QueryMatchContext` instance?

But looks like I probably could change it to have the lifetime of `self.file_run.from_file_run_context_instance_provider`, whatever I don't think I'm going to dive back into this right now